src/screens/Library.jsx

```diff
@@ -0,0 +1,101 @@
+import React, {useState, useEffect} from "react";
+import likesStyles from "./Likes.module.css";
+import profileStyles from "./Profile.module.css";
+import Card from "../components/Card";
+import { API } from "../static/Services";
+import Loader from '../components/Loader';
+import buttonStyles from "../components/Button.module.css";
+
+const defaultState = {
+    labs: [],
+    candidates: [],
+    loading: <Loader />,
+    isCompanyMode: true
+}
+
+const style = {
+    fontSize: '32px'
+}
+
+const Library = (props) =>  {
+    const [state, setState] = useState(defaultState);
+    const role = state.isCompanyMode ? 'Company' : 'Candidate';
+
+    useEffect( () => {
+        API.fetchAllLabs().then(labs => {
```

Comment:
This is how you should do it, call useEffect, inside it make the API call, and then set the state to the result of the call. However, I think it would be more readable with async/await, will show you tomorrow how to use that inside useEffect.

Reply:
Also, another thing you could do, is to fetch data based on the role. Right now you are fetching data for candidates, and labs, even though you don't use them both. A solution for this would be in the "data layer" to fetch one or the other, based on the role.